pandas/io/feather_format.py

```diff
@@ -0,0 +1,90 @@
+""" feather-format compat """
+
+from pandas import DataFrame, RangeIndex, MultiIndex, Int64Index
+from pandas.compat import range
+from pandas.lib import infer_dtype
+
+
+def _try_import():
+    # since pandas is a dependency of feather
+    # we need to import on first use
+
+    try:
+        import feather
+    except ImportError:
+
+        # give a nice error message
+        raise ImportError("the feather-format library is not installed\n"
+                          "you can install via conda\n"
+                          "conda install feather-format -c conda-forge")
+    return feather
+
+
+def to_feather(df, path):
+    """
+    Write a DataFrame to the feather-format
+
+    Parameters
+    ----------
+    df : DataFrame
+    path : string
+        File path
+    """
+    if not isinstance(df, DataFrame):
+        raise ValueError("feather only support IO with DataFrames")
+
+    feather = _try_import()
+
+    # validate that we have only a default index
+    # raise on anything else as we don't serialize the index
+
+    if not isinstance(df.index, (RangeIndex, Int64Index)):
+        raise ValueError("feather does not serializing {} "
+                         "for the index; you can .reset_index()"
+                         "to make the index into column(s)".format(
+                             type(df.index)))
+
+    if not df.index.equals(RangeIndex.from_range(range(len(df)))):
+        raise ValueError("feather does not serializing a non-default index "
+                         "for the index; you can .reset_index()"
+                         "to make the index into column(s)")
+
+    if isinstance(df.columns, MultiIndex):
+        raise ValueError("feather does not serializing a MultiIndex "
+                         "for the columns")
+
+    # validate that we do not have any non-string object dtypes
+    # as these 'work', but will not properly de-serialize
+    valid_types = set(['string', 'unicode'])
```

Comment:
this provides a check for this serialization issue: https://github.com/wesm/feather/issues/240